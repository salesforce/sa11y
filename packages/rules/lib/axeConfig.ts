/*
 * Copyright (c) 2020, salesforce.com, inc.
 * All rights reserved.
 * SPDX-License-Identifier: BSD-3-Clause
 * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
 */

import { RunOptions } from 'axe-core';

/**
 * AxeConfig is limited to subset of options that we need and use in this library
 */
export interface AxeConfig extends RunOptions {
    runOnly: {
        type: 'rule';
        values: string[];
    };
    resultTypes: ['violations'];
}

/**
 * Returns config to be used in axe.run() with given rules
 *
 * @param rules - List of rules to be used in the config
 * @returns AxeConfig with formatted rules
 */
export function getAxeConfig(rules: string[]): AxeConfig {
    return {
        runOnly: {
            type: 'rule',
            values: rules,
        },
        resultTypes: ['violations'],
    };
}
