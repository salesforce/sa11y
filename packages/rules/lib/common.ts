/*
 * Copyright (c) 2020, salesforce.com, inc.
 * All rights reserved.
 * SPDX-License-Identifier: BSD-3-Clause
 * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
 */

// Common config across all rulesets
export const common = {
    runOnly: {
        type: 'rule',
    },
    resultTypes: ['violations'],
};
