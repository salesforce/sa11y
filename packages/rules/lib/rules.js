'use strict';

module.exports = rules;

function rules() {
    // TODO
}
