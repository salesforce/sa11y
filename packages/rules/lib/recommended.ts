/*
 * Copyright (c) 2020, salesforce.com, inc.
 * All rights reserved.
 * SPDX-License-Identifier: BSD-3-Clause
 * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
 */

import { getAxeConfig } from './axeConfig';

const rules = [
    'area-alt',
    'aria-allowed-attr',
    'aria-hidden-body',
    'aria-hidden-focus',
    'aria-input-field-name',
    'aria-required-attr',
    'aria-required-children',
    'aria-required-parent',
    'aria-roledescription',
    'aria-roles',
    'aria-toggle-field-name',
    'aria-valid-attr-value',
    'aria-valid-attr',
    'audio-caption',
    'autocomplete-valid',
    'blink',
    'button-name',
    'bypass',
    'color-contrast',
    'definition-list',
    'dlitem',
    'document-title',
    'duplicate-id-active',
    'duplicate-id-aria',
    'duplicate-id',
    'empty-heading',
    'focus-order-semantics',
    'form-field-multiple-labels',
    'frame-title',
    'heading-order',
    'html-has-lang',
    'html-lang-valid',
    'html-xml-lang-mismatch',
    'image-alt',
    'image-redundant-alt',
    'input-button-name',
    'input-image-alt',
    'label-title-only',
    'label',
    'link-name',
    'list',
    'listitem',
    'marquee',
    'meta-refresh',
    'page-has-heading-one',
    'role-img-alt',
    'scope-attr-valid',
    'scrollable-region-focusable',
    'server-side-image-map',
    'svg-img-alt',
    'tabindex',
    'td-has-header',
    'td-headers-attr',
    'th-has-data-cells',
    'valid-lang',
    'video-caption',
];

export const recommended = getAxeConfig(rules);
