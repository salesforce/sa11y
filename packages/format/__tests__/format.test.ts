/*
 * Copyright (c) 2020, salesforce.com, inc.
 * All rights reserved.
 * SPDX-License-Identifier: BSD-3-Clause
 * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
 */

import 'global-jsdom/lib/register'; // https://github.com/dequelabs/axe-core/blob/develop/doc/API.md#required-globals
import * as axe from 'axe-core';
import { a11yResultsFormatter } from '..';

afterEach(() => {
    document.body.innerHTML = ''; // reset dom body
});

describe('a11y Results Formatter', () => {
    it('should format a11y issues as expected', async () => {
        document.body.innerHTML = `<html>
                                    <body>
                                     <a href="#"></a>
                                    </body>
                                   </html>`;
        await axe.run(document).then((results) => {
            expect(results).toBeDefined();
            expect(a11yResultsFormatter(results.violations)).toMatchSnapshot();
        });
    });
});
