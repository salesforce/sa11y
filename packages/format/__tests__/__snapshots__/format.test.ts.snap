// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`a11y Results Formatter should format a11y issues as expected 1`] = `
"Page must have means to bypass repeated blocks (bypass): html
  - More info: https://dequeuniversity.com/rules/axe/3.5/bypass?application=axeAPI

Documents must have <title> element to aid in navigation (document-title): html
  - More info: https://dequeuniversity.com/rules/axe/3.5/document-title?application=axeAPI

<html> element must have a lang attribute (html-has-lang): html
  - More info: https://dequeuniversity.com/rules/axe/3.5/html-has-lang?application=axeAPI

Document must have one main landmark (landmark-one-main): html
  - More info: https://dequeuniversity.com/rules/axe/3.5/landmark-one-main?application=axeAPI

Links must have discernible text (link-name): a
  - More info: https://dequeuniversity.com/rules/axe/3.5/link-name?application=axeAPI

Page must contain a level-one heading (page-has-heading-one): html
  - More info: https://dequeuniversity.com/rules/axe/3.5/page-has-heading-one?application=axeAPI"
`;
